monitor/data_monitoring.py

```diff
@@ -160,23 +153,32 @@ def generate_report(self, tracking: AnonymousTracking, days_back: Optional[int]
                 elementary_output_json_file:
             elementary_output_json_file.write(dumped_output_data)
 
-        webbrowser.open_new_tab('file://' + html_path)
+        if should_open_browser:
+            webbrowser.open_new_tab('file://' + html_path)
         self.execution_properties['report_end'] = True
         self.execution_properties['success'] = self.success
         return self.success, html_path
 
-    def send_report(self, elementary_html_path: str) -> bool:
-        if os.path.exists(elementary_html_path):
-            file_uploaded_successfully = self.slack_client.upload_file(
-                channel_name=self.slack_channel_name,
-                file_path=elementary_html_path,
-                message=SlackMessageSchema(text="Elementary monitoring report")
-            )
-            if not file_uploaded_successfully:
+    def send_report(self, html_path: str) -> bool:
+        if self.slack_client:
+            send_succeded = self.slack_client.send_report(self.config.slack_channel_name, html_path)
+            self.execution_properties['sent_to_slack_successfully'] = send_succeded
+            if not send_succeded:
                 self.success = False
-        else:
-            logger.error('Could not send Elementary monitoring report because it does not exist')
-            self.success = False
+
+        if self.s3_client:
+            send_succeded = self.s3_client.send_report(html_path)
+            self.execution_properties['sent_to_s3_successfully'] = send_succeded
+            if not send_succeded:
+                self.success = False
+
+        if self.gcs_client:
+            send_succeded = self.gcs_client.send_report(html_path)
+            self.execution_properties['sent_to_gcs_successfully'] = send_succeded
+            if not send_succeded:
+                self.success = False
+
+        self.execution_properties['success'] = self.success
```

Comment:
I think you should still return self.success as well at the end of the method

Reply:
I do though.
```py
        self.execution_properties['success'] = self.success
        return self.success
```